### PR TITLE
fix:1447 Included github image configuration, docker configuration & Pagination to the API in API Catalog

### PR DESCRIPTION
**Pull request by riginoommen**
# Fixes 1447

# Explain the fix

- Included the Pagination for the API
- Included the API documentation, Docker Configuration & Github Image configuration

## Does this PR introduce a breaking change
No

## Screenshots

<details>
<summary>View Screenshots</summary>
image
</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?

### CODE REVIEW

File: packages/api-catalog-service/src/types.d.ts [hunk 1]
```diff
@@ -100,6 +100,10 @@ interface NamespaceInput {
     updatedBy?: Maybe<Scalars[ 'String' ]>;
 }
 
+interface ListNamespacesArgs {
+    limit?: Maybe<Scalars['Int']>;
```
Comment (ghanlohar) [inline packages/api-catalog-service/src/types.d.ts:104]: What is the benefit of using the `Scalers` interface. Isn't it another abstraction?

**Review by ghanlohar — CHANGES_REQUESTED** [review]

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]

File: packages/api-catalog-service/src/types.d.ts (hunk 1 above)
Comment (riginoommen) [reply packages/api-catalog-service/src/types.d.ts]: updated

File: .github/workflows/build-api-catalog-service.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+name: Build API Catalog Service Image
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches:
+      - 'master'
+      - 'release'
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply .github/workflows/build-api-catalog-service.yml]: updated

File: .github/workflows/build-api-catalog-service.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      - name: Generate a short tag using GITHUB_SHA
+        run: echo "SHA_TAG=$(echo $GITHUB_SHA | cut -c 1-6)" >> $GITHUB_ENV
+      - if: github.event.inputs.tags == '' && startsWith(github.ref, 'refs/heads/master')
+        name: Generate QA tag
+        run: echo "TAGS=latest,qa" >> $GITHUB_ENV
+      - if: github.event.inputs.tags == '' && startsWith(github.ref, 'refs/heads/release')
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply .github/workflows/build-api-catalog-service.yml]: updated

**Review by deshmukhmayur — APPROVED** [review]

**Review by ghanlohar — APPROVED** [review]

**Pull request merged**